pkg/util/testing/client.go

```diff
@@ -27,7 +27,7 @@ import (
 )
 
 func NewFakeClient(objs ...client.Object) client.Client {
-	return NewClientBuilder().WithObjects(objs...).Build()
+	return NewClientBuilder().WithObjects(objs...).WithStatusSubresource(objs...).Build()
```

Comment:
Since controller-runtime v0.15.0, we need to register objects to fakeClient for StatusSubresource.

> Testing: Fake client status handling (https://github.com/kubernetes-sigs/controller-runtime/pull/2259)
Added a new WithStatusSubresource option and pre-populating it with
all in-tree resources that have a status subresource
Update and Patch methods now don't change the status for any such
resource anymore
The status clients Update and Patch methods now only change the status
for any such resource

https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0